### PR TITLE
fix duplicate engine entries

### PR DESCRIPTION
**Pull request by mjkkirschner**
TEMP PR until ironPythondotnet goes into DynamoDS/Dynamo

### CODE REVIEW

File: src/Libraries/PythonNodeModelsWpf/PythonNode.cs [hunk 1]
```diff
@@ -61,13 +93,7 @@ public void CustomizeView(PythonNode nodeModel, NodeView nodeView)
...
             learnMoreItem.Click += OpenPythonLearningMaterial;
 
-            var availableEngineNames = PythonEngineManager.Instance.AvailableEngines.Select(x => x.Name).ToList();
-            // Add the serialized Python Engine even if it is missing (so that the user does not see an empty slot)
-            if (!availableEngineNames.Contains(nodeModel.EngineName))
-            {
-                availableEngineNames.Add(nodeModel.EngineName);
-            }
-            availableEngineNames.ForEach(x => AddPythonEngineToMenuItems(x));
+            PythonNodeUtils.GetEngineNames(nodeModel).ForEach(x => AddPythonEngineToMenuItems(x));
```
Comment (aparajit-pratap) [inline src/Libraries/PythonNodeModelsWpf/PythonNode.cs:96]: What if the menu already contains the engine name(s) returned from `GetEngineNames`?

File: src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs [hunk 2]
```diff
@@ -1256,7 +1256,7 @@ internal void HideUnhideNamespace(bool hide, string library, string namespc)
         private void AddPythonEnginesOptions()
         {
             var options = new ObservableCollection<string> { Res.DefaultPythonEngineNone };
-            foreach (var item in PythonEngineManager.Instance.AvailableEngines)
+            foreach (var item in PythonEngineManager.Instance.AvailableEngines.GroupBy(x=>x.Name).FirstOrDefault().ToList())
```
Comment (pinzart90) [inline src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs:1259]: were there multiple python engines with the same name ?
like 2 versions of Python2 at the same time?

File: src/Libraries/PythonNodeModelsWpf/PythonNode.cs (hunk 1 above)
Comment (mjkkirschner) [reply src/Libraries/PythonNodeModelsWpf/PythonNode.cs:96]: I've tried to stop that from happening, but I will add another check in `AddPythonEngineToMenuItems` as well just to be sure.

File: src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs (hunk 2 above)
Comment (mjkkirschner) [reply src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs:1259]: yes, thats exactly the bug.

Comment (pinzart90) [reply src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs:1259]: so if we will support assembly isolation, then we could have the same pacakge but different versions loaded at the same time
So should we enforce uniqueness by appending the version to the engine name (internally at least)?
In the UI we could display the same name but tooltip would clarify which version. Or something like that ..

File: src/Libraries/PythonNodeModelsWpf/PythonNode.cs (hunk 1 above)
Comment (mjkkirschner) [reply src/Libraries/PythonNodeModelsWpf/PythonNode.cs:96]: done

**Pull request closed without merging**